framework/Optimizers/gradients/FiniteDifference.py

```diff
@@ -68,28 +74,50 @@ def evaluate(self, opt, grads, infos, objVar):
       @ Out, direction, dict, versor (unit vector) for gradient direction
     """
     gradient = {}
+    A=[]
+    #print("This is grads",grads[0].keys())
+    #print("This is opt",opt)
+    #aaa
+    #print("These are")
     for g, pt in enumerate(grads):
+      x, y= enumerate(grads)
+
       info = infos[g]
       delta = info['delta']
       activeVar = info['optVar']
-      lossDiff = np.atleast_1d(mathUtils.diffWithInfinites(pt[objVar], opt[objVar]))
-      #if delta != 0:
-      grad = (lossDiff) / delta
-      #else:
-      #  grad = 0
```

Comment:
I don't feel like "evaluate" is long enough to warrant further separation; it's entire purpose is to evaluate the derivative, so moving the derivative evaluation to a separate function seems superfluous.